Computer_vision_workloads/Semantic_segmentation/Scaling_batch_inference.ipynb

```diff
@@ -2124,29 +2127,56 @@
    ]
   },
   {
-   "cell_type": "markdown",
-   "id": "0d8a0b7a-3cff-4aea-9b92-c28200560728",
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b4e512ca",
    "metadata": {},
+   "outputs": [],
    "source": [
-    "### Shutdown Ray runtime"
+    "### YOUR CODE HERE ###"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "fc3aae2c-491f-41a9-bd67-09395f398890",
+   "cell_type": "markdown",
+   "id": "fececa99",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "ray.shutdown()"
+    "**Solution**"
    ]
   },
   {
-   "cell_type": "markdown",
-   "id": "72cdc4c3-5c2e-489a-9382-d8ca46a26a1f",
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "14709a03",
    "metadata": {},
+   "outputs": [],
    "source": [
-    "Disconnect the worker, and terminate processes started by `ray.init()`."
+    "### SAMPLE IMPLEMENTATION ###\n",
+    "\n",
+    "from transformers import AutoConfig, AutoModelForSemanticSegmentation, AutoFeatureExtractor\n",
+    "from transformers.pipelines import pipeline\n",
+    "from ray.train.huggingface import HuggingFacePredictor\n",
+    "\n",
+    "# specify HuggingFace model and feature extractor\n",
+    "model_checkpoint = \"nvidia/segformer-b0-finetuned-ade-512-512\"\n",
+    "feature_extractor_checkpoint = \"nvidia/segformer-b0-finetuned-ade-512-512\"\n",
+    "\n",
+    "feature_extractor = AutoFeatureExtractor.from_pretrained(feature_extractor_checkpoint)\n",
+    "model_config = AutoConfig.from_pretrained(model_checkpoint)\n",
+    "model = AutoModelForSemanticSegmentation.from_config(model_config)\n",
+    "\n",
+    "# instantiate Predictor with type of task, model, and feature extractor\n",
+    "predictor = HuggingFacePredictor(\n",
+    "    pipeline=pipeline(\n",
+    "        task=\"image-segmentation\", model=model, feature_extractor=feature_extractor\n",
+    "    )\n",
+    ")\n",
+    "\n",
+    "# list of image data -> pandas dataframe\n",
+    "pd_data = pd.DataFrame(data)\n",
```

Comment:
You seem to re-use data from ActorsPool part -> better to create dataset here again.

Can we work with numpy arrays here?

Reply:
New version is using the Ray Datasets `dataset` from Part 8 to perform batch inference. The input is a Ray Dataset, and the output is a Ray Dataset (i.e. `predictions_dataset`)